scalyr_agent/util.py

```diff
@@ -1269,3 +1272,87 @@ def close(self):
             """Closes the channel to the server.
             """
             pass
+
+
+def parse_array_of_strings(strlist):
+    """Convert comma-separated string list into an ArrayOfStrings
+
+    Accepts the following string representations.
+    ['a', 'b', 'c']
+    ["a", "b", "c"]
+    'a', 'b', 'c'
+    "a", "b", "c"
+    a, b, c
+
+    @param strlist: list to be converted
+    @return: None if strlist is empty, else return a JsonArray of strings
+    @raise TypeError if element_type is specified and conversion of any element fails
+    """
+    if not strlist:
+        return None
+    strlist = strlist.strip()
+
+    # Remove surrounding square brackets
+    if strlist[0] == '[' and strlist[-1] == ']':
+        strlist = strlist[1:-1]
+
+    # Extract elements, removing any surrounding quotes (Single-quotes are illegal JSON. Double quotes will be added).
+    elems = []
+    items = strlist.split(',')
+    for elem in items:
+        elem = elem.strip()
+        if len(elem) == 0:
+            continue
+        if elem[0] == r"'" or elem[0] == r'"' and elem[-1] == elem[0]:
+            elem = elem[1:-1]
+        if len(elem) == 0:
+            continue
+        elems.append(elem)
+
+    return ArrayOfStrings(*elems)
+
+
+def get_config_from_env(param_name, custom_env_name=None, convert_to=None):
+    """Returns the environment variable value for a config param
+
+    If a custom environment variable name is defined, use it instead of prepending 'SCALYR_'.
+
+    @param param_name: Config param name (may be global or module-level)
+    @custom_env_name: Custom environment variable name
+    @expected_type: if not None, will convert and validate to this type.  Otherwise, will leave as string.
+    @return: 3-tuple containing (param_name, env string val, converted value)
+        Where converted_value is an object representing the converted environment value.
+    @raise TypeError if cannot be converted to expected_type
+    """
+    key = custom_env_name
+    if not key:
+        key = 'SCALYR_%s' % param_name
+
+    key = key.upper()
+    strval = os.getenv(key)
+
+    if not strval:
+        key = key.lower()
+        strval = os.getenv(key)
+
+    if not strval or convert_to is None:
+        return key, strval, None
+
+    try:
+        if convert_to in {unicode, str}:
+            return key, strval, convert_to(strval)
+
+        # Special-case handling for ArrayOfStrings
```

Comment:
It seems there's a overlap between this code and the conversion code you modified in the `configuration.py`.  Is there a way to move this method in there so that it can rely on the `_perform_conversion` method?

Reply:
I will investigate.